service/history/queues/executable.go

```diff
@@ -84,6 +83,12 @@ type (
 	ExecutorWrapper interface {
 		Wrap(delegate Executor) Executor
 	}
+
+	// TerminalErrors are errors which cannot be retried and should not be scheduled again.
+	// Tasks should be enqueued to a DLQ immediately if an error implements this interface.
+	TerminalTaskError interface {
+		IsTerminalTaskError()
```

Comment:
Common practice is for IsXXX function to return a boolean value. Even if you are not going to implement the function. When other engineers see IsXXX they expect true or false.